File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {    
        // for(int i=0; i < nums.size(); i++) {
        //     int theNum = nums[i];
        //         for(int j = i + 1; j < nums.size(); j++ ) {
        //             if (nums[j] == theNum) {
        //                 return true;
        //             }
        //         }
        // }
        // return false;
        unordered_map<int, int> map;

        for(int i = 0; i < nums.size(); i++) {
            if (map.find(nums[i]) == map.end()) {
                map[nums[i]] = 1;
            } else {
                return true;
            }
        }
        return false;
    }
};

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        theMap = {}

        for num in nums:
            if num not in theMap:
                theMap[num] = 1
            else:
                theMap[num] += 1

        for count in theMap.values():
            if count > 1 :
                return True
        return False